tests/fsyacc/main.fs

```diff
@@ -0,0 +1,76 @@
+open Tree
+open System.IO
+open Microsoft.FSharp.Quotations
+open Microsoft.FSharp.Compatibility.OCaml
+open FSharp.Text.Lexing
+let tokenize = ref false
+
+let usage = [ "--tokens", Arg.Set tokenize, "tokenize the first file and exit" ]
+
+let mutable inputs = []
+
+Arg.parse usage (fun x -> inputs <- inputs @ [x]) "test... <filename> <filename>\nTests that all inputs give equivalent syntac trees"
+
+let createLexBuffer (a:Expr<'a->_>) (x:FileStream) : 'a =
+    (if typeof<'a> = typeof<LexBuffer<char>> then
+        x |> StreamReader |> LexBuffer<_>.FromTextReader :> obj
+    elif typeof<'a> = typeof<LexBuffer<byte>> then
+        x |> BinaryReader |> LexBuffer<_>.FromBinaryReader :> obj
+    else
+        failwith "Pies")
+    :?> _
+
+if inputs = [] then
+    Printf.eprintf "at least one input should be given\n";
+try 
+  let results =
+    inputs
+    |> List.map
+      (fun filename -> 
+        use is = File.OpenRead filename
+        let lexbuf = createLexBuffer <@ TestLexer.token @> is
+        if !tokenize then
+          while true do 
+            Printf.eprintf "tokenize - getting one token\n"
+            let t = TestLexer.token lexbuf
+            Printf.eprintf "tokenize - got %s, now at char %d\n" (TestParser.token_to_string t) (lexbuf.StartPos).pos_cnum
+            match t with 
+            | TestParser.EOF -> exit 0
+            | TestParser.IDENT s -> 
```

Comment:
It's hard to see in the diff but this `match` in the `main.fs` (originally `main.ml`) comes from `main-unicode.ml` and allows testing the exact names of the lexed IDENT tokens.